File: source-code/Server/backend/src/main/java/com/iasdietideals24/backend/mapstruct/dto/ProfiloDto.java
package com.iasdietideals24.backend.mapstruct.dto;

import com.iasdietideals24.backend.mapstruct.dto.shallows.AccountShallowDto;
import com.iasdietideals24.backend.mapstruct.dto.utilities.AnagraficaProfiloDto;
import com.iasdietideals24.backend.mapstruct.dto.utilities.LinksProfiloDto;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
public class ProfiloDto {

    private String nomeUtente;

    private byte[] profilePicture;

    private AnagraficaProfiloDto anagrafica;

    private LinksProfiloDto links;

    private Set<AccountShallowDto> accountsShallow;
}

File: source-code/Client/frontendandroid/app/src/main/java/com/iasdietideals24/dietideals24/utilities/dto/ProfiloDto.kt
package com.iasdietideals24.dietideals24.utilities.dto

import com.iasdietideals24.dietideals24.utilities.dto.shallows.AccountShallowDto
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDate

@Getter
@Setter
@NoArgsConstructor
class ProfiloDto {
    private val nomeUtente: String? = null

    private val profilePicture: ByteArray? = null

    private val nome: String? = null

    private val cognome: String? = null

    private val dataNascita: LocalDate? = null

    private val areaGeografica: String? = null

    private val biografia: String? = null

    private val linkPersonale: String? = null

    private val linkInstagram: String? = null

    private val linkFacebook: String? = null

    private val linkGitHub: String? = null

    private val linkX: String? = null

    private val accountsShallow: Set<AccountShallowDto>? = null
}
